### ISSUE TITLE
Length aware pagination

### ISSUE THREAD

**Issue by jimgwhit** [laravel/docs#4997]
In the docs, you have:

Quote
When manually creating a paginator instance, you should manually "slice" the array of results you pass to the paginator. If you're unsure how to do this, check out the array_slice PHP function.
unquote

Please change this comment to reflect:

* It's alright to slice small results 
* Large results should be calculated with offset 

Example only:
```
    public function indexGet()
    {
        $page = Request::input('page', '1');
        $search = Request::input('search', '');
        $sch = $search . "%";
        $perpage = "5";
        $offset = ($page - 1) * $perpage;
        $krows = DB::select('select COUNT(dogid) as count from dc_dogs');
        $numrows = $krows[0]->count;
        Session::put('search', $search); // later use
        $pagingQuery = "LIMIT {$offset}, {$perpage}";
        $sql = "SELECT * FROM dc_dogs WHERE `dogname` LIKE :sch " . $pagingQuery;
        $sth = DB::getPdo()->prepare($sql);
        $params = [':sch' => $sch];
        $sth->execute($params);
        $sth->execute();
        $quy = $sth->fetchAll(\PDO::FETCH_OBJ);
        $dogs = new \Illuminate\Pagination\LengthAwarePaginator($quy, $numrows, $perpage);
       
        $qs = ['search' => $search, 'page' => $page];
        $title = 'Dogs';
        return view('dog.indexg', compact('dogs', 'title', 'qs'));
    }
```
Of course several ways to use
```
$dogs = DB::table('dc_dogs')
                        ->skip($offset)->take($perpage)->get();
        $pageclass = 'custompager';
        $pagelinks = LengthPager::makeLengthAware($dogs, $numrows, $perpage, ['t1' => $t1], $pageclass);
```
But please see this:

https://laracasts.com/discuss/channels/guides/length-aware-paginator

and

https://christophersax.com/2016/custom-lengthawarepaginator-in-laravel/

Bottom line, slice is fine on smaller data results, not huge.  Newbies see that slice in the docs, and well think it's the only way.

**Comment by driesvints** [comment]
Please just send in a PR if you think this needs changing.
